models/DataObject/ClassDefinition/CustomLayout.php

```diff
@@ -462,20 +466,27 @@ public function setName($name)
     }
 
     /**
-     * @return int
+     * @return bool|int (will be bool)
      */
+    #[\ReturnTypeWillChange]
     public function getDefault()
     {
         return $this->default;
     }
 
     /**
-     * @param int $default
+     * @param bool|int $default
      *
      * @return $this
      */
     public function setDefault($default)
     {
+        if(is_int($default)){
+            trigger_deprecation('pimcore/pimcore', '10.6',
+                sprintf('Passing int to %s is deprecated and will be removed in Pimcore 11.' .
+                    'Use bool instead.', __FUNCTION__));
+        }
+
         $this->default = (int)$default;
```

Comment:
We should allow setting the bool values. So the users get now deprecation warnings if they set the correct value.
```suggestion
        if (is_bool($default)) {
            $this->default = $default;
        } else {
            trigger_deprecation('pimcore/pimcore', '10.6',
                sprintf('Passing int to %s is deprecated and will be removed in Pimcore 11.' .
                    'Use bool instead.', __FUNCTION__));
            $this->default = (int)$default;
        }
```

Reply:
@blankse, we only output a deprecation warning if the user passes an integer ...

https://github.com/pimcore/pimcore/blob/24c3db3667f44ca7e82096d2183c2fd3c2a3534d/models/DataObject/ClassDefinition/CustomLayout.php#L484-L488

Reply:
But the method would also be executed when a User load the custom layout and he cant set a bool

Reply:
but if a user cant set `bool` your solution would also print a deprecation or? since you can only pass `int` and `bool`

Reply:
With my change the user can set a bool. See line 485